### PR TITLE
New devices and support for manufacturer / model specification

### PR DESCRIPTION
**Pull request by timlaing**
Added support for:

- Diivoo Dual Smart Water Timer
- Generic Tuya Smart Water Timer
- Specification of Manufacturer & Model in yaml configuration

Bugfixes:

- Execution during startup for creation of task not on main thread.

### CODE REVIEW

**Comment by fcrozat** [comment]
it might be worth looking also at my own PR: https://github.com/make-all/tuya-local/pull/851

File: custom_components/tuya_local/config_flow.py [hunk 1]
```diff
@@ -162,6 +162,10 @@ async def async_step_user(self, user_input=None):
                 errors["base"] = "connection"
 
         schema = {
+            vol.Required(
```
[the remark this answers is not in the document]
Comment (timlaing) [reply custom_components/tuya_local/config_flow.py:165]: Thanks for the advice. I will revert this change, and push an updated version once I have all the tests passing.

File: custom_components/tuya_local/device.py [hunk 2]
```diff
@@ -605,7 +614,7 @@ def get_key_for_value(obj, value, fallback=None):
         return keys[values.index(value)] if value in values else fallback
 
 
-def setup_device(hass: HomeAssistant, config: dict):
+async def setup_device(hass: HomeAssistant, config: dict):
```
[the remark this answers is not in the document]
Comment (timlaing) [reply custom_components/tuya_local/device.py:621]: I was trying to fix an error running this component in the home assistant dev container. Unfortunatly it did not work so reverting this change.

File: custom_components/tuya_local/device.py [hunk 3]
```diff
@@ -128,18 +133,22 @@ def unique_id(self):
...
+
+        return device_info
 
     @property
     def has_returned_state(self):
         """Return True if the device has returned some state."""
         return len(self._get_cached_state()) > 1
 
-    def actually_start(self, event=None):
+    async def actually_start(self, event=None):
```
[the remark this answers is not in the document]
Comment (timlaing) [reply custom_components/tuya_local/device.py]: As per the other async comment, now reverting this change.

**Pull request closed without merging**